rust_src/src/frames.rs

```diff
@@ -58,6 +73,16 @@ impl LispFrameRef {
     pub fn set_selected_window(&mut self, window: LispObject) {
         self.selected_window = window.to_raw();
     }
+
+    #[inline]
+    pub fn is_visible(self) -> bool {
+        unsafe { fget_visible(self.as_ptr()) }
+    }
+
+    #[inline]
+    pub fn is_iconified(self) -> bool {
+        unsafe { LispObject::from_bool(fget_iconified(self.as_ptr()) as BoolBF).is_not_nil() }
```

Comment:
You should be able to do this without turning it into a LispObject along the way.